### PR TITLE
Issue81

### PR DESCRIPTION
**Pull request by mariwalt**
Achtung: Dieser Branch hat fixing_leonard als Basis. Daher werden alle Änderungen aus fixing_leonard auch hier angezeigt)

Was hat sich hier geändert: 
1) Emails lassen sich nun deutlich schneller runterladen/anzeigen
2) Löschen von Emails ist möglich

Achtung: wenn ihr nun eine Email löscht, dann ist diese endgültig weg! Also seid vorsichtig beim Testen!!! 
Warum ist das so? Das Löschen mittels Imap geht über zwei Schritte:
1) Man gibt einer Emails das Flag \Delete. (Bis hierhin passiert also gar nichts auf dem Server - nichts wird gelöscht, verschoben oder so. Es wird nur ein Flag gesetzt.)
2) Nun sendet man das Kommando Expunge an der Server. Damit werden alle Emails mit den Flag \Delete im aktuellen Ordner endgültig gelöscht.

Eine gute Lösung wäre also, dass wir zwischen den beiden Schritten die Emails mit den Flag \Delete in den Thrash-Ordner verschieben. Und dort führen wir dann das Expunge-Kommando aus. Da wir aber noch nicht Emails verschieben können, müssen wir uns mit der derzeitigen Lösung erstmal zufrieden geben.

### CODE REVIEW

**Comment by davidhennemann** [comment]
Will review tomorrow. Sorry was a little bit busy at weekend 😄

File: packages/IMAPClient-Core.package/ICEndPoint.class/instance/parseFetchFlags.for..st [hunk 1]
```diff
@@ -0,0 +1,13 @@
+protocol parse
+parseFetchFlags: aHeaderChunk for: anEmail
+
+	((aHeaderChunk at: 1) includesSubstring: '\Seen')
```
Comment (davidhennemann) [inline packages/IMAPClient-Core.package/ICEndPoint.class/instance/parseFetchFlags.for..st]: I would suggest: `anEmail seen: ((aHeaderChunk at: 1) includesSubstring: '\Seen')`. No need for the conditional logic :)

File: packages/IMAPClient-Core.package/ICEndPoint.class/instance/parseFetchFlags.for..st [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+		ifTrue: [anEmail seen: true] 
+		ifFalse: [anEmail seen: false].
+		
+	((aHeaderChunk at: 1) includesSubstring: '\Flagged')
```
Comment (davidhennemann) [inline packages/IMAPClient-Core.package/ICEndPoint.class/instance/parseFetchFlags.for..st]: see above

**Review by davidhennemann — CHANGES_REQUESTED** [review]
Nearly everthing is fine. (Only a few points --> see reviews)
Can you merge develop into your branch?
I will fix tests for you :)

**Comment by mariwalt** [comment]
?ups, those are good changes!


I merged develop in it and pushed it!

________________________________

File: packages/IMAPClient-Core.package/ICFileReadWriter.class/instance/get.in..st [hunk 3]
```diff
@@ -8,5 +8,7 @@ get: aFile in: aFolder
 		header: (self createHeaderFrom: (jsonFile at: 'header'));
 		body: (jsonFile at: 'body');
 		messageID: (jsonFile at: 'messageID');
+		seen: (jsonFile at: 'seen');
```
Comment (davidhennemann) [inline packages/IMAPClient-Core.package/ICFileReadWriter.class/instance/get.in..st]: `seen` and `flagged` are header informations, those will be loaded within header: `(self createHeaderFrom: (jsonFile at: 'header'));`. There are only present in the jsonFile 'header' field, setting them here, will result in always setting them to `nil`. Couldnt have figured this out without the test...wow maybe tests are helpful at all :D will fix this within my commit, which will fix the tests. :)

**Review by davidhennemann — CHANGES_REQUESTED** [review]
Okay, i fixed the tests :) Just had to add some flags to the test data. But i noticed an issue. 
Steps to reproduce:
1. Mark a Mail as Unseen
image
2. Close Mail Client, and reopen it.
3. Dont update from Server, just use the already loaded version. Now the Mail wont display the [UNSEEN] Flag, but seems still to be unread, because right clicking it, gives me the option to `Mark as seen`.
image

I think you need to add the [UNSEEN] in front of the name if mails get loaded from storage to fix this :) 
Even pressing the update button, wont get the [UNSEEN] Flag back...its just stays as it is. But the Content Menu still says `Mark as seen`. So i think its just a rendering issue.

**Comment by mariwalt** [comment]
It's fixed now?.

________________________________

**Comment by davidhennemann** [comment]
Yeah, it works flawlessly now :) Should we merge this one?

**Review by davidhennemann — APPROVED** [review]

**Comment by davidhennemann** [comment]
Okay, still found an issue, but i am not sure, whether we need an extra issue for that. 

1. Mark an Mail as unseen.
2. Read mail on another device.
3. Mail is still marked as unseen, even after `Update`. The Update Button should also update flags for existing mails, shoudnt it? 

But we can create a sepeate issue for this, because we need to figure out a nice way, how to update already loaded mails.... What do you think?

**Comment by mariwalt** [comment]
?In that case pressing update will load the locally stored email. And the locally stored email does not know about any changes on the other device. For similar reason I initially never stored and loaded the flags locally, but always loaded from the server (thus, there was never a reason to store them). But this caused bad performance, so I changed it in this issue again...

Maybe we should create another issue for this problem and solve it on another branch, but merge this branch already in develop.

________________________________

**Comment by davidhennemann** [comment]
Okay, sounds good. We need to figure out a smart way how to update locally saved mails from the server, witout running into a bad performance. Created Follow-Up-Issue: https://github.com/hpi-swa-teaching/IMAPClient/issues/100

**Pull request merged**